src/libappimage/core/impl/TraversalType2.cpp

```diff
@@ -25,256 +23,352 @@ extern "C" {
-    if (ret != 0)
-        throw IOError("symlink error at " + target);
-}
-
-istream& TraversalType2::read() {
-
-    // resolve symlinks if any
-    if (!resolveSymlink(&currentInode))
-        throw IOError("symlink resolution error");
-
-    // create a streambuf for reading the inode contents
-    auto tmpBuffer = new StreambufType2(&fs, &currentInode, 1024);
-
-    // replace buffer of the istream
-    entryIStream.rdbuf(tmpBuffer);
-
-    // replace and drop old buffer
-    entryStreamBuf.reset(tmpBuffer);
-
-    return entryIStream;
-}
-
-bool TraversalType2::resolveSymlink(sqfs_inode* inode) {
-    sqfs_err err;
-    bool found = false;
-
-    sqfs_inode rootInode;
-    err = sqfs_inode_get(&fs, &rootInode, rootInodeId);
-    if (err != SQFS_OK)
-        return false;
-
-    // Save visited inode numbers to prevent a infinite loop in case of cycles between symlinks.
-    // A cycle may occur when by example: a (file) -> (links to) b and b -> c and c -> a
-    std::set<__le32> inodesVisited;
-    inodesVisited.insert(inode->base.inode_number);
-
-    while (inode->base.inode_type == SQUASHFS_SYMLINK_TYPE || inode->base.inode_type == SQUASHFS_LSYMLINK_TYPE) {
-        // Read symlink
-        size_t size;
-        // read twice, once to find out right amount of memory to allocate
-        err = sqfs_readlink(&fs, inode, NULL, &size);
-        if (err != SQFS_OK)
-            return false;
-
-        char symlinkTargetPath[size];
-        // then to populate the buffer
-        err = sqfs_readlink(&fs, inode, symlinkTargetPath, &size);
-        if (err != SQFS_OK)
-            return false;
-
-        // lookup symlink target path
-        *inode = rootInode;
-        err = sqfs_lookup_path(&fs, inode, symlinkTargetPath, &found);
-
-        if (!found || err != SQFS_OK)
-            return false;
-
-        // check if we fell into a symlinks cycle
-        auto ret = inodesVisited.insert(inode->base.inode_number);
-        if (!ret.second)
-            throw PayloadIteratorError("Symlinks loop found ");
-
-    }
-
-    return true;
-}
-
-string TraversalType2::getEntryLinkTarget() const {
-    return currentEntryLink;
-}
-
-appimage::core::PayloadEntryType TraversalType2::readEntryType() const {
-    // squashfs traversal follows a DFS pattern but directories are "visited" twice, when they are reached and when they
-    // are left. This check is to properly identify the second scenario.
-    if (trv.dir_end)
-        return PayloadEntryType::DIR;
-
-    switch (trv.entry.type) {
-        case SQUASHFS_REG_TYPE:
-        case SQUASHFS_LREG_TYPE:
-            return PayloadEntryType::REGULAR;
-
-        case SQUASHFS_SYMLINK_TYPE:
-        case SQUASHFS_LSYMLINK_TYPE:
-            return PayloadEntryType::LINK;
-
-        case SQUASHFS_DIR_TYPE:
-        case SQUASHFS_LDIR_TYPE:
-            return PayloadEntryType::DIR;
-
-        default:
-            return PayloadEntryType::UNKNOWN;
+namespace appimage {
```

Comment:
Why do you make such _huge_ changes in a single PR? It's nearly impossible to compare those side-by-side now on GitHub...

I can't see why it was necessary to make this change. IIRC I already suggested using such a syntax instead of the `using namespace` used above, but you explained that this way you'd save lots of whitespace. I don't see any reason why you would have to change that now.

Reply:
Reverted the part regarding namespaces, to ease review.

I'm aware you suggested using this namespaces usage style before. Also, we use nested namespaces in the major part of the `cpp` files in our project. As I was doing a big refactor it seemed to be a good chance to also change it.